src/SaleorAuthClient.ts

```diff
@@ -150,6 +151,10 @@ export class SaleorAuthClient {
   fetchWithAuth: Fetch = async (input, init) => {
     const refreshToken = this.storageHandler?.getRefreshToken();
 
+    if (!this.accessToken) {
+      this.accessToken = cookie.parse(document.cookie).token;
```

Comment:
Let's also delete the cookie here.
```suggestion
      this.accessToken = cookie.parse(document.cookie).token;
      document.cookie = cookie.serialize("token", "", { expires: new Date(0) });
```